ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/rlpx/RlpxAgent.java

```diff
@@ -374,6 +390,21 @@ private void handleIncomingConnection(final PeerConnection peerConnection) {
     enforceConnectionLimits();
   }
 
+  private boolean doTheFractionOfRemoteConnectionsAllowsNewOne() {
+    final int remotelyInitiatedConnectionsCount =
+        Math.toIntExact(
+            connectionsById.values().stream()
+                .filter(RlpxConnection::isActive)
+                .filter(RlpxConnection::initiatedRemotely)
+                .count());
+    if (remotelyInitiatedConnectionsCount == 0) {
+      return true;
+    }
+    final double fractionRemoteConnections =
+        (double) remotelyInitiatedConnectionsCount / (double) getConnectionCount();
```

Comment:
I think this should be against the max allowed connection count, not a ratio against the current set of connection.  @mbaxter, what's your opinion?

Reply:
yes i think it would be better, i update the computation

Reply:
Done.